client/src/containers/UserProfile/UserProfile.js

```diff
@@ -1,41 +1,56 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import { fetchEvents } from '../../actions';
+import { Button } from 'reactstrap';
+import { fetchUserProfile } from '../../actions';
 import Profile from './../../components/UserProfile/UserProfile';
 import { AttendeeType } from '../../types/attendee-types';
+import { Link } from 'react-router-dom';
+import jwtDecode from 'jwt-decode';
+import { store } from '../../../src';
 
 class UserProfile extends Component {
   constructor(props) {
     super(props);
-    this.fetchEvents();
+    this.fetchUserProfile();
   }
 
-  fetchEvents() {
+  fetchUserProfile() {
+    let userId = this.props.match.params.user_id;
     const { dispatch } = this.props;
-    dispatch(fetchEvents());
+    dispatch(fetchUserProfile(userId));
   }
-  getUserDetails(): AttendeeType {
-    const {
-        event_id: eventId = null,
-        attendee_id: attendeeId = null,
-      } = this.props.match.params,
-      { events = [] } = this.props;
-    let event, attendee;
-    event = events.find(event => event.id == eventId);
-    if (event && event.attendees)
-      attendee = event.attendees.find(att => (att.id === attendeeId));
 
-    return attendee ? attendee : null;
+  getUserDetails(): AttendeeType {
+    let user = this.props.userProfile;
+    return user ? user : null;
   }
+
+  showEditButton = user => {
+    const { token } = store.getState().userState;
+    const payloadUserId = jwtDecode(token).user_id;
```

Comment:
@taleymirza @JavaEria why do we need to decode this? We should be able to get the info the api